### PR TITLE
Enabling Jaeger E2E Tests

### PR DESCRIPTION
**Pull request by anajosep**
Fixed the failures in Kind AT tests with K8S 1.20 where cronjobs are not available in `v1` and instead available only in `v1beta`. Verified the stability of the tests using push triggerred acceptance tests and upgrade paths tests.

### CODE REVIEW

File: tests/e2e/jaeger/helidon/jaeger_helidon_test.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+	metrics.Emit(t.Metrics.With("undeployment_elapsed_time", time.Since(start).Milliseconds()))
+})
+
+var _ = t.Describe("Jaeger Operator", Label("f:jaeger.helidon-workload"), func() {
+	t.Context("after successful installation", func() {
+		// GIVEN the Jaeger Operator is enabled and a sample application is installed,
+		// WHEN we check for traces for that service,
+		// THEN we are able to get the traces
+		WhenJaegerOperatorEnabledIt("should have a verrazzano-monitoring namespace", func() {
```
Comment (anders-swanson) [inline tests/e2e/jaeger/helidon/jaeger_helidon_test.go:151]: description of IT method is wrong

File: tests/e2e/jaeger/helidon/jaeger_helidon_test.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+					}
+				}
+				return tracesFound, nil
+			}).WithPolling(shortPollingInterval).WithTimeout(shortWaitTimeout).Should(BeTrue())
+		})
+
+		// GIVEN the Jaeger Operator component is enabled,
+		// WHEN a sample application is installed,
+		// THEN the traces are found in OpenSearch Backend
+		WhenJaegerOperatorEnabledIt("should have running pods", func() {
```
Comment (anders-swanson) [inline tests/e2e/jaeger/helidon/jaeger_helidon_test.go:178]: Description of IT method is wrong

File: tests/e2e/pkg/jaeger.go [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+	var cronJobs []v1.CronJob
+	cronJobList, err := clientset.BatchV1().CronJobs(namespace).List(context.TODO(), listOptions)
+	if err != nil {
+		Log(Error, fmt.Sprintf("Failed to list v1/cronjobs in namespace %s: %v", namespace, err))
+		return cronJobs, err
+	}
+	return cronJobList.Items, nil
+}
+
+// listV1Beta1CronJobNames lists the cronjob under batch/v1beta1 api version for k8s version <= 1.20
```
Comment (anders-swanson) [inline tests/e2e/pkg/jaeger.go:312]: Is there ever a time this test is running on k8s 1.19 or lower?

File: tests/e2e/pkg/keycloak.go [hunk 4]
```diff
@@ -216,7 +216,9 @@ func (c *KeycloakRESTClient) DeleteUser(userRealm string, userID string) error {
 		return fmt.Errorf("invalid response")
 	}
 	defer response.Body.Close()
-	if response.StatusCode != 200 {
+	// all responses in the 200 range are acceptable
+	// in practice, this call returns 204 (No Content) for success
+	if response.StatusCode < 200 || response.StatusCode >= 300 {
```
Comment (anders-swanson) [inline tests/e2e/pkg/keycloak.go:221]: why don't we check for 204 code if that is what is expected?

File: tests/testdata/jaeger/helidon/helidon-tracing-comp.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          containers:
+            - name: hello-helidon-container
+              image: "ghcr.io/verrazzano/example-helidon-greet-app-v1:1.0.0-1-20220513221156-7da0d32"
+              env:
+                - name: "TRACING_SERVICE"
+                  value: "hello-helidon"
+                - name: "TRACING_PORT"
+                  value: "9411"
+                - name: "TRACING_HOST"
+                  value: "jaeger-operator-jaeger-collector.verrazzano-monitoring.svc.cluster.local"
```
Comment (anders-swanson) [inline tests/testdata/jaeger/helidon/helidon-tracing-comp.yaml:34]: If you are using sidecar injection, Jaeger should use localhost?  You may need to specify protocol if UDP is being expected by the agent.

Comment (anajosep) [reply tests/testdata/jaeger/helidon/helidon-tracing-comp.yaml:34]: Tried using localhost and UDP port, but the tests are failing.

File: tests/e2e/pkg/keycloak.go (hunk 4 above)
Comment (anajosep) [reply tests/e2e/pkg/keycloak.go:221]: To make it generic as all 2XX codes are success codes.

File: tests/e2e/pkg/jaeger.go (hunk 3 above)
Comment (anajosep) [reply tests/e2e/pkg/jaeger.go:312]: Yes, push triggerred tests run against 1.20 which was failing in my earlier attempt to merge this PR.

File: tests/e2e/jaeger/helidon/jaeger_helidon_test.go (hunk 2 above)
Comment (anajosep) [reply tests/e2e/jaeger/helidon/jaeger_helidon_test.go]: Fixed

File: tests/e2e/jaeger/helidon/jaeger_helidon_test.go (hunk 1 above)
Comment (anajosep) [reply tests/e2e/jaeger/helidon/jaeger_helidon_test.go]: fixed

File: tests/testdata/jaeger/helidon/helidon-tracing-comp.yaml (hunk 5 above)
Comment (ayaranga) [reply tests/testdata/jaeger/helidon/helidon-tracing-comp.yaml:34]: In the subsequent Pr, at least remove `"sidecar.jaegertracing.io/inject": "true"` . Having sidecar but connecting to collector is confusing.

**Review by ayaranga — APPROVED** [review]

**Review by anders-swanson — APPROVED** [review]

**Pull request merged**